paddle/fluid/operators/reshape_op.cc

```diff
@@ -99,7 +127,7 @@ and target shape = [1, 4], the reshape operator will transform
 the tensor X into a 2-D tensor: [[1, 2, 3, 4]]
 
 One dimension in the target shape can be set -1, representing that its
-size is unknown. In this case, the real dimension will be infered from 
+size is unknown. In this case, the real dimension will be infered from
 the original shape of Input(X) and other dimensions in the target shape.
 )DOC");
```

Comment:
Need to explain when the dimension can be set to 0 (necessary) and in-place reshape (optional) in the doc.

Reply:
Done.